### PR TITLE
Dev/more clippy

### PR DESCRIPTION
**Pull request by xnorpx**
make clippy happy

### CODE REVIEW

**Review by algesten — CHANGES_REQUESTED** [review]
Looks good, thanks! One thing to discuss.

File: rtp/src/rtcp/sdes.rs [hunk 1]
```diff
@@ -103,8 +103,8 @@ impl Sdes {
 
         let pad = 4 - tot % 4;
         if pad < 4 {
-            for i in 0..pad {
-                buf[i] = SdesType::END as u8;
+            for val in buf.iter_mut().take(pad) {
+                *val = SdesType::END as u8;
```
[the remark this answers is not in the document]
Comment (xnorpx) [reply rtp/src/rtcp/sdes.rs:107]: yeah, I wasn't sure about it either. I agree it loses readability. You might add safety as the iterator might do bounds checking. Will disable the lint on these

[the remark this answers is not in the document]
Comment (algesten) [reply rtp/src/rtcp/sdes.rs:107]: Thanks!

File: ice/src/pair.rs [hunk 2]
```diff
@@ -137,7 +137,7 @@ impl CandidatePair {
             (remote_prio, local_prio)
         };
 
-        2_u64.pow(32) * g.min(d) as u64 + 2 * g.max(d) as u64 + if g > d { 1 } else { 0 }
+        2_u64.pow(32) * g.min(d) as u64 + 2 * g.max(d) as u64 + u64::from(g > d)
```
Comment (algesten) [inline ice/src/pair.rs:140]: let's not change this

File: ice/src/agent.rs [hunk 3]
```diff
@@ -404,7 +404,7 @@ impl IceAgent {
                 ServerReflexive => 32_767,
                 Relayed => 16_383,
             };
-            x - if ip.is_ipv6() { 0 } else { 1 }
+            x - u32::from(!ip.is_ipv6())
```
Comment (algesten) [inline ice/src/agent.rs:407]: let's not change this

**Review by algesten — CHANGES_REQUESTED** [review]

**Comment by algesten** [comment]
This is in main now.

**Pull request closed without merging**